packages/core/src/Stores/client-store.js

```
@@ -2767,21 +2767,30 @@ export default class ClientStore extends BaseStore {
         const matchingSubscription = this.subscriptions?.[name]?.[key]?.sub;
         if (matchingSubscription) return { key, subscription: matchingSubscription };
 
-        const subscription = WS?.subscribe({
-            [name]: 1,
-            subscribe: 1,
-            ...(payload ?? {}),
-        });
+        try {
+            const subscription = await WS?.subscribe({
+                [name]: 1,
+                subscribe: 1,
+                ...(payload ?? {}),
+            });
 
-        this.addSubscription(name, key, subscription);
-        return { name, key, subscription };
+            this.addSubscription(name, key, subscription);
+            return { key, subscription };
+        } catch (e) {
+            // error handling needed
+            return {};
```

can we return the error so consumers can handle it?

---

Answered below